third_party/move/move-core/types/src/value.rs

```diff
@@ -465,6 +465,13 @@ impl MoveStructLayout {
             },
         }
     }
+
+    pub fn signer() -> Self {
```

Comment:
for reference/documentation purposes, I would add into signer.move a comment like this:

```
    // // at runtime: 
    // enum signer {
    //     Master{account: address},
    //     Permissioned{account: address, permissions_address}
    // }
    // // for bcs serialization:
    // struct signer {
    //     account: address,
    // }
```